File: solutions/062.unique-paths/unique-paths.js
/**
 * @param {number} m
 * @param {number} n
 * @return {number}
 */
const uniquePaths = (m, n) => {
  let dp = []

  for (let i = 0; i < m; i++) {
    dp[i] = new Array(n)
  }

  for (let i = 0; i < m; i++) {
    dp[i][0] = 1
    for (let j = 0; j < n; j++) {
      dp[0][j] = 1

      if (i > 0 && j > 0) {
        dp[i][j] = dp[i - 1][j] + dp[i][j - 1]
      }
    }
  }

  return dp[m - 1][n - 1]
}

File: solutions/062.unique-paths/unique-paths.py
class Solution(object):
    def uniquePaths(self, m, n):
        """
        :type m: int
        :type n: int
        :rtype: int
        c[i][j] = c[i -1]c[j] + c[i][j-1]
        """
        # init
        grid = []
        for i in range(m):
            grid.append([0] * n)

        grid[0][0] = 1

        for i in range(m):
            for j in range(n):
                if i == 0 and j == 0:
                    continue
                grid[i][j] = grid[i-1][j] + grid[i][j-1]

        return grid[-1][-1]
